code/datums/loadout/loadout_manager.dm

```diff
@@ -6,18 +6,20 @@
 	/**
 	 * List of all loadouts. Format is list(list(loadout_job, loadout_name))
 	 */
-	var/loadouts_data = list()
+	var/list/loadouts_data = list()
 	/// The host of the loadout_manager, aka from which loadout vendor are you managing loadouts
 	var/loadout_vendor
 	/// The version of the loadout manager
 	var/version = CURRENT_LOADOUT_VERSION
 
 ///Remove the data of a loadout from the loadouts list
 /datum/loadout_manager/proc/delete_loadout(mob/user, loadout_name, loadout_job)
+	var/list/new_loadouts_data = list()
 	for(var/loadout_data in loadouts_data)
-		if(loadout_data[1] == loadout_job && loadout_data[2] == loadout_name)
-			loadouts_data -= loadout_data
-			return
+		if(loadout_data[1] != loadout_job || loadout_data[2] != loadout_name)
+			new_loadouts_data += list(loadout_data)
+	loadouts_data = new_loadouts_data
+	user.client?.prefs.save_loadout_list(loadouts_data, CURRENT_LOADOUT_VERSION)
```

Comment:
Just use cut()?

Reply:
this is doing the same, but better, since it deletes duplicate loadouts rather than just the first one.